File: KASPAR GUI/src/runners/ActionRunner.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package runners;

import data.Action;
import data.Pose;
import data.Robot;
import data.Sequence;
import data.Sound;
import gui.GuiLogger;
import java.util.logging.Level;

/**
 *
 * @author nathan
 */
public class ActionRunner {
    
    private Robot robot;
    
    public ActionRunner(Robot robot)
    {
        this.robot = robot;
    }

    public static boolean isValid(Action action) {
        //TODO: check for valid
        return action != null;
    }

    public void execute(Action action) {
        if (action instanceof Sequence) {
            new SequenceRunner(this.robot).execute((Sequence)action);
        } else if (action instanceof Pose) {
            new PoseRunner(this.robot).execute((Pose)action);
        } else if (action instanceof Sound) {
            new SoundRunner(this.robot).execute((Sound)action);
        } else {
            GuiLogger.getLogger().log(Level.SEVERE, "Could not determine action type for " + action.toString(), action);
        }
    }
}

File: KASPAR_python/Robot/runner/actionRunner.py
# import logging
# from Data.Model.Action import Sequence, Group, Sound, Pose
# from sequenceRunner import SequenceRunner
# from poseRunner import PoseRunner
# from soundRunner import SoundRunner
# from groupRunner import GroupRunner
# 
# class ActionRunner(object):
# 
#     def __init__(self, robot):
#         self._robot = robot;
#         self._logger = logging.getLogger(__name__)
# 
#     @staticmethod
#     def isValid(action):
#         #TODO: check for valid
#         return action != None
# 
#     def execute(self, action):
#         if type(action) == Sequence:
#             SequenceRunner(self._robot).execute(action)
#         elif type(action) == Pose:
#             PoseRunner(self._robot).execute(action)
#         elif type(action) == Sound:
#             SoundRunner(self._robot).execute(action)
#         elif type(action) == Group:
#             GroupRunner(self._robot).execute(action)
#         else:
#             self._logger.critical("Could not determine action type for %s" % (action))
